code/API_definitions/qod-provisioning.yaml

```
@@ -57,6 +57,16 @@ info:
     ### Restrictions for tokens without an associated authenticated identifier:
 
     - For scenarios which do not have a single device identifier associated to the token during the authentication flow, e.g. 2-legged access tokens, the `device` object MUST be provided in the API request. This ensures that the device identification is explicit and valid for each API call made with these tokens.
+
+    # Multi-SIM scenario handling
+
+    In multi-SIM scenarios where more than one mobile device is associated with a phone number (e.g. a smartphone with an associated smartwatch), it might not be possible to uniquely identify the device to which the QoS provisioning should apply from that phone number. If the phone number is used as the device identifier when provisioning in a multi-SIM scenario, the API may respond with an error, or may apply the provisioning to a device other than the intended device.

```

Another possibility is to apply the service to **all** the devices associated to the phone number:

```suggestion
    In multi-SIM scenarios where more than one mobile device is associated with a phone number (e.g. a smartphone with an associated smartwatch), it might not be possible to uniquely identify the device to which the QoS provisioning should apply from that phone number. If the phone number is used as the device identifier when provisioning in a multi-SIM scenario, the API may respond with an error, or may apply the provisioning to all the devices associated with a phone number.
```

---

I'm here on the side of @jlurien option: either an error or all the devices associated with the phone number. Applying the provisioning to a device other than the intended device would be an erroneous behavior which the API consumer can not detect. That leaves for an API provider who doesn't want to apply the provisioning to all devices still the option to respond with an error (btw: which one?).

---

I was thinking of the scenario where an API provider allows the use of secondary MSISDNs for provisioning, but the API consumer mistakenly uses the primary MSISDN when intending to provision one of the secondary devices. Thus provisioning is applied to the primary device (smartphone) when their intention was to provision, say, the tablet device, **but they didn't use the API correctly**.

I think we have two options here:
- Allow the use of secondary MSISDNs for individual device provisioning, with sufficient documentation being provided to the API consumer to avoid them using the wrong MSISDN; or
- Provision for all devices when the primary MSISDN is used, otherwise provision for the secondary device associated with the secondary MSISDN.

The issues with this second solution are:
- the primary device cannot be provisioned individually
- not all devices may support the requested QoS profile
- every implementation will be required to do a multi-SIM check for every API invocation, which is a complication I could well live without for a first commercial version of this API

I appreciate that some implementations may use this second option, so will update the text accordingly. But until the general issue on multi-SIM handling in Commonalities is resolved, I think multi-SIM handling for the QoD APIs should remain somewhat "proprietary".